website/pages/docs/testing-best-practices.mdx

```diff
@@ -0,0 +1,134 @@
+---
+title: Testing Best Practices
+sidebarTitle: Testing Best Practices
+---
+
+# Testing Best Practices
+
+As your GraphQL server grows, so does the risk of regressions, inconsistencies, 
+and slow development feedback. A thoughtful testing strategy helps you catch 
+problems early and ship with confidence—without overwhelming your team with 
+redundant or brittle tests.
+
+This guide outlines practical testing patterns for GraphQL servers, 
+including schema safety, test coverage, data management, performance, and 
+continuous integration.
+
+## Schema stability
+
+Your schema is a contract with every client and consumer. Changes should 
+be intentional and reviewed.
+
+### Best practices
+
+- Use snapshot tests to catch unintended schema changes  
+  – Tool: [`jest-serializer-graphql-schema`](https://github.com/kamilkisiela/jest-serializer-graphql-schema)  
+  – Example:
+    ```ts
+    expect(printSchema(schema)).toMatchSnapshot();
+    ```
```

Comment:
```suggestion
    ```
    - Consider sorting the schema (`graphql.lexicographicSortSchema(schema)`) to ensure
      a stable order of your schemas types, fields and arguments in the snapshots.
```